packages/shared/src/utils/mounted/mounted.js

```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+export const useIsMounted = () => {
+    const is_mounted = React.useRef(false);
+
+    React.useEffect(() => {
+        is_mounted.current = true;
+
+        return () => (is_mounted.current = false);
+    }, []);
+    return is_mounted;
```

Comment:
Could we return the value of `is_mounted.current` rather than the `ref` itself? This is so we can reference it as `is_mounted` instead of `is_mounted.current` in the future.

```JS
const Component = () => {
    const is_mounted = useIsMounted();
    
    React.useEffect(() => {
        if (is_mounted) {
            // Some logic to execute when mounted.
            // Old version is "if (is_mounted.current)"
        }
    }, []);
}
```

Reply:
So I tried returning `is_mounted.current` but it didn't work it would return `is_mounted.current = false" so I was playing around with this and turns out, if I return `is_mounted` and use it as is in other components (without the `.current`), it works. Is this alright?